hera_pspec/pspecdata.py

```diff
@@ -2171,17 +2199,32 @@ def pspec_run(dsets, filename, groupname=None, dset_labels=None, dset_pairs=None
         # get redundant baseline groups
         bls = None
 
+    # Construct dataset pairs to operate on
+    Ndsets = len(dsets)
+    if dset_pairs is None:
+        dset_pairs = list(itertools.combinations(range(Ndsets), 2))
+
+    if dset_labels is None:
```

Comment:
Should we also check for duplicate dset labels? That could screw things up if it ever happens.